linalg/hypre.cpp

```diff
@@ -128,6 +128,13 @@ HypreParVector::HypreParVector(ParFiniteElementSpace *pfes)
    own_ParVector = 1;
 }
 
+void HypreParVector::WrapHypreParVector(hypre_ParVector *y)
+{
+   x = y;
+   _SetDataAndSize_();
+   own_ParVector = 0;
+}
```

Comment:
Before overwriting the current contents of the `HypreParVector` one has to destroy the old one. In this case, we need to do what the destructor `HypreParVector::~HypreParVector` does explicitly:
```
   if (own_ParVector) { hypre_ParVectorDestroy(x); }
```
and also reset the base class `Vector` by calling `Vector::Destroy`.